### PR TITLE
history for ensembles and final layer option. 

### PR DESCRIPTION
**Pull request by odiazib**
1)	Adding history attribute in learner class; however, I only include training history of last iteration.  
2)	Adding “sum” option in final_layer in ResNets. This operation will reduce one dimension in the output tensor. 
3)	Adding relative error computation for ensembles.

### CODE REVIEW

File: quinn/ens/learner.py [hunk 1]
```diff
@@ -14,6 +14,8 @@ class Learner():
         best_model (torch.nn.Module): The best trained PyTorch NN module.
         trained (bool): Whether the module is trained or not.
         verbose (bool): Whether to be verbose or not.
+        history (list of doubles): List containing training history of last iteration, namely, [fepoch, loss_trn, loss_trn_full, loss_val]
```
Comment (ksargsyan) [inline quinn/ens/learner.py]: Please reword to list[double]

Comment (ksargsyan) [reply quinn/ens/learner.py]: Also, I don't understand why this attribute is needed and why should it be called history if it is only the last iteration and not the full history.

File: quinn/nns/nnbase.py [hunk 2]
```diff
@@ -85,16 +85,16 @@ def fit(self, xtrn, ytrn, **kwargs):
             **kwargs (dict): Keyword arguments.
 
         Returns:
-            torch.nn.Module: Best trained instance.
+            dict: Dictionary of the results. Keys 'best_fepoch', 'best_epoch', 'best_loss', 'best_nnmodel', 'history'.
```
Comment (ksargsyan) [inline quinn/nns/nnbase.py]: Remind me why we need to return dict and not the best model. I am afraid several codes relying on these won't be backward compatible.

File: quinn/utils/metrics.py [hunk 3]
```diff
@@ -3,6 +3,19 @@
 
 import numpy as np
 
+def rel_l2_ens(predictions, targets):
```
Comment (ksargsyan) [inline quinn/utils/metrics.py:6]: This needs to be commented a lot better. It seems it assumes certain structure/dimension of the arrays. There must be a more pythonic way of doing this.

**Review by ksargsyan — CHANGES_REQUESTED** [review]

File: quinn/nns/nnbase.py (hunk 2 above)
Comment (odiazib) [reply quinn/nns/nnbase.py]: I reverted this change.

File: quinn/ens/learner.py (hunk 1 above)
Comment (odiazib) [reply quinn/ens/learner.py]: I decided to remove this attribute. We can access to the training history using: `learner.nnmodel.history`

File: quinn/utils/metrics.py (hunk 3 above)
Comment (odiazib) [reply quinn/utils/metrics.py:6]: Done.

**Pull request merged**